include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -188,6 +188,39 @@ __reduce_over_group(_Args... __args)
 #endif
 }
 
+template <typename... _Args>
+constexpr auto
+__any_of_group(_Args... __args)
+{
+#if _ONEDPL_SYCL2020_COLLECTIVES_PRESENT
```

Comment:
Are we sure that either of APIs is available on all the compilers we are supporting today?

Reply:
I will check on this and get back to you. These APIs are supported across our compiler configurations we use in CI, but I will verify this is the case for everything we support.

Reply:
I have looked into this and the oneAPI group extensions for the algorithms I have added were provided in the `2021.1` DPC++ compiler alongside the scan and reduce group algorithms which we already use. The SYCL 2020 group algorithms were provided in `2021.4`. I believe it is therefore safe to add the rest of these group algorithms into `sycl_defs.h`.